packages/doenetml-worker-rust/lib-doenetml-core/src/core/dast/macro_expand.test.rs

```diff
@@ -0,0 +1,240 @@
+use assert_json_diff::assert_json_eq;
+
+use super::*;
+use crate::test_utils::*;
+
+#[test]
+fn macros_get_expanded_to_their_referents() {
```

Comment:
I'd like to write a bunch more tests nailing down the edge cases of this expander to make sure we have the behavior well specified. Would you want them in this PR? I can try to think through different edge cases.

Reply:
Oh, it looks like you have more macro resolve tests, below.